hydradx/notebooks/Omnipool/vault/ImpermanentLossAnalysis.ipynb

```diff
@@ -0,0 +1,224 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import os\n",
+    "import glob\n",
+    "import random\n",
+    "sys.path.append('../../..')\n",
+    "\n",
+    "from model import processing\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "from model import run\n",
+    "from model import plot_utils as pu\n",
+    "from model.amm.omnipool_amm import OmnipoolState, cash_out_omnipool, value_assets, usd_price, lrna_price\n",
+    "from model.amm.agents import Agent\n",
+    "from model.amm.trade_strategies import omnipool_arbitrage, invest_all, price_sensitive_trading\n",
+    "from model.amm.global_state import GlobalState, fluctuate_prices, historical_prices\n",
+    "\n",
+    "# same seed, same parameters = same simulation result\n",
+    "random.seed(42)\n",
+    "# price_list = processing.import_binance_prices(['BTC', 'ETH', 'DOT'], start_date='Jan 1 2023', days = 120)\n",
+    "\n",
+    "assets = {\n",
+    "    'HDX': {'usd price': 0.05, 'weight': 0.7999},\n",
+    "    'USD': {'usd price': 1, 'weight': 0.20},\n",
+    "    'TKN': {'usd price': 1, 'weight': 0.0001}\n",
+    "}\n",
+    "\n",
+    "lrna_price_usd = 0.07\n",
+    "initial_omnipool_tvl = 10000000\n",
+    "liquidity = {}\n",
+    "lrna = {}\n",
+    "\n",
+    "for tkn, info in assets.items():\n",
+    "    liquidity[tkn] = initial_omnipool_tvl * info['weight'] / info['usd price']\n",
+    "    lrna[tkn] = initial_omnipool_tvl * info['weight'] / lrna_price_usd\n",
+    "\n",
+    "initial_state = GlobalState(\n",
+    "    pools={\n",
+    "        'omnipool': OmnipoolState(\n",
+    "            tokens={\n",
+    "                tkn: {'liquidity': liquidity[tkn], 'LRNA': lrna[tkn]} for tkn in assets\n",
+    "            },\n",
+    "            asset_fee=0,\n",
+    "            lrna_fee=0,\n",
+    "            # withdrawal_fee=False\n",
```

Comment:
Does this mean that the withdrawal_fee is *on* in this analysis?

Reply:
There's no withdrawal fee, but that's because OmnipoolArchiveState doesn't include that attribute. That's probably something that should be addressed? It certainly makes a case like this unclear.

Reply:
So is the withdrawal fee on or off in this analysis?